### ISSUE TITLE
Pages without <head>

### ISSUE THREAD

**Issue by Armael** [aantron/dream-serve#3]
I got very confused for a moment when trying out dream-serve, not understanding why it would not work on my dummy (and I assume invalid) webpage with just `<h1> hello </h1>`. The answer being (as I found out looking at the code) that dream-serve will not inject the websocket script if the webpage does not contain a `<head>`.

I think it would be helpful to do one of the following:
- document that in the README
- emit a logging message when encountering a webpage with no head (saying that it does not get instrumented)
- add a `<head>` with the script even for pages that do not have one originally

**Comment by aantron** [comment]
Hey, thanks for the report. This is actually just an instance of https://github.com/tmattio/dream-livereload/issues/1. In detail, dream-serve is parsing the HTML as fragments, which is Lambda Soup's default. It just needs to set the context to document, and the parser will automatically insert `<head>`. I'll fix it shortly.

**Comment by aantron** [comment]
(And your HTML is valid &mdash; `<head>` is optional even for complete HTML documents)

**Comment by aantron** [comment]
It should be working now. Also, double check that you have Dream 1.0.0~alpha2, not ~alpha1. When I installed Dream just now for the upgrade of dream-serve, opam picked ~alpha1 because of the conflict with the current broken ppxlib (https://github.com/ocaml-ppx/ppxlib/issues/221). I had to do `opam install dream.1.0.0~alpha2`. I updated both the Dream README and the forum post with this install command to minimize surprises.
